src/control/vendor/github.com/daos-stack/go-spdk/spdk/src/nvme_control.c

```diff
@@ -101,8 +101,8 @@ attach_cb(void *cb_ctx, const struct spdk_nvme_transport_id *trid,
 		exit(1);
 	}
 
+	spdk_pci_addr_parse(&entry->pci_addr, trid->traddr);
 	entry->ctrlr = ctrlr;
-	entry->tr_addr = trid->traddr;
```

Comment:
So this is the root cause of why the device can't be correctly selected, right?

Reply:
no, it was the logic in get_controller and incorrect use of Pointer